### PR TITLE
doc: Add MBA documentation and update doc for CAT

### PR DESCRIPTION
**Pull request by vijaydhanraj**
This patch updates CAT documentation as well as adds
documentation support for newly added MBA feature.

Signed-off-by: Vijay Dhanraj <vijay.dhanraj@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by ClaudZhang1995** [comment]
hi, @vijaydhanraj ,the doc pr will make doc fail .It is fail log,please update.
New errors/warnings found, please fix them:
==============================================

/root/workspace/acrn-hypervisor-premerge-build-image/acrn-hypervisor/build/doc/rst/release_notes/release_notes_0.8.rst:66: WARNING: undefined label: using_cat_up2 (if the link has no caption the label must precede a section header)
/root/workspace/acrn-hypervisor-premerge-build-image/acrn-hypervisor/build/doc/rst/release_notes/release_notes_1.0.rst:224: WARNING: undefined label: using_cat_up2 (if the link has no caption the label must precede a section header)

**Comment by vijaydhanraj** [comment]
> # hi, @vijaydhanraj ,the doc pr will make doc fail .It is fail log,please update.
> New errors/warnings found, please fix them:
> /root/workspace/acrn-hypervisor-premerge-build-image/acrn-hypervisor/build/doc/rst/release_notes/release_notes_0.8.rst:66: WARNING: undefined label: using_cat_up2 (if the link has no caption the label must precede a section header)
> /root/workspace/acrn-hypervisor-premerge-build-image/acrn-hypervisor/build/doc/rst/release_notes/release_notes_1.0.rst:224: WARNING: undefined label: using_cat_up2 (if the link has no caption the label must precede a section header)



> # hi, @vijaydhanraj ,the doc pr will make doc fail .It is fail log,please update.
> New errors/warnings found, please fix them:
> /root/workspace/acrn-hypervisor-premerge-build-image/acrn-hypervisor/build/doc/rst/release_notes/release_notes_0.8.rst:66: WARNING: undefined label: using_cat_up2 (if the link has no caption the label must precede a section header)
> /root/workspace/acrn-hypervisor-premerge-build-image/acrn-hypervisor/build/doc/rst/release_notes/release_notes_1.0.rst:224: WARNING: undefined label: using_cat_up2 (if the link has no caption the label must precede a section header)

Fixed and pushed the changes

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: doc/tutorials/rdt_configuration.rst [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+                      .os_config = {
+                              .name = SOS_VM_CONFIG_OS_NAME,
+                      },
+              },
+      };
+
+.. note::
+   In ACRN, Lower CLOS always means higher priority (clos 0 > clos 1 > clos 2>...clos n). So care should be taken to program the VMs CLOS accordingly.
+
+#. Careful consideration should be made in assigning VCPU affinity. In cache isolation configuration, not only need to isolate CAT-capable caches, but need to isolate lower-level caches as well. In the following example, logical processor #0 and #2 share L1 and L2 caches. In this case, do not assign LP #0 and LP #2 to different VMs that need to do cache isolation. Assign LP #1 and LP #3 with similar consideration.
```
Comment (gvancuts) [inline doc/tutorials/rdt_configuration.rst]:
```suggestion
#. Careful consideration should be made in assigning vCPU affinity. In cache isolation configuration, not only need to isolate CAT-capable caches, but need to isolate lower-level caches as well. In the following example, logical processor #0 and #2 share L1 and L2 caches. In this case, do not assign LP #0 and LP #2 to different VMs that need to do cache isolation. Assign LP #1 and LP #3 with similar consideration.
```

File: doc/tutorials/rdt_configuration.rst [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+          L2Cache L#1 (size=256KB linesize=64 ways=4 Inclusive=0)
+            L1dCache L#1 (size=32KB linesize=64 ways=8 Inclusive=0)
+              L1iCache L#1 (size=32KB linesize=64 ways=8 Inclusive=0)
+                Core L#1 (P#1)
+                  PU L#2 (P#1)
+                  PU L#3 (P#3)
+
+#. Similary bandwidth control is per-core (not per LP), so max delay values of per-LP CLOS is applied to the core. If HT is turned on, don’t place high priority threads on sibling LP running lower priority threads.
+
+#. Build the ACRN hypervisor and copy the artifact ``acrn.efi`` to the
```
Comment (gvancuts) [inline doc/tutorials/rdt_configuration.rst]: We have had problems in the past with people building and updating the `acrn.efi` hypervisor but not the user-space tools (esp. `acrn-dm`). While we do not want to add full instructions on how to do that, let's at least put in a note to make sure people understand this needs to be done too.

**Comment by vijaydhanraj** [comment]
> Very good documentation! A few comments but overall looks good!!

Thanks, @gvancuts. I have addressed all the below comments and pushed the changes.

**Pull request merged**